econml/validate/drtester.py

```diff
@@ -0,0 +1,796 @@
+
+        Parameters
+        ----------
+        Xval: (n_val x k) matrix or vector of length n
+            Features used in nuisance models for validation sample
+        Dval: vector of length n_val
+            Treatment assignment of validation sample. Control status must be minimum value. It is recommended to have
+            the control status be equal to 0, and all other treatments integers starting at 1.
+        yval: vector of length n_val
+            Outcomes for the validation sample
+        Xtrain: (n_train x k) matrix or vector of length n, default ``None``
+            Features used in nuisance models for training sample
+        Dtrain: vector of length n_train, default ``None''
+            Treatment assignment of training sample. Control status must be minimum value. It is recommended to have
+            the control status be equal to 0, and all other treatments integers starting at 1.
+        ytrain: vector of length n_train, defaul ``None``
+            Outcomes for the training sample
+
+        Returns
+        ------
+        self, with added attributes for the validation treatments (Dval), treatment values (tmts),
+        number of treatments excluding control (n_treat), boolean flag for whether training data is provided
+        (fit_on_train), doubly robust outcome values for the validation set (dr_val), and the DR ATE value (ate_val).
+        If training data is provided, also adds attributes for the doubly robust outcomes for the training
+        set (dr_train) and the training treatments (Dtrain)
+
+        """
+
+        self.Dval = Dval
+
+        # Unique treatments (ordered, includes control)
+        self.tmts = np.sort(np.unique(Dval))
+
+        # Number of treatments (excluding control)
+        self.n_treat = len(self.tmts) - 1
+
+        # Indicator for whether
+        self.fit_on_train = (Xtrain is not None) and (Dtrain is not None) and (ytrain is not None)
+
+        if self.fit_on_train:
+            # Get DR outcomes in training sample
+            reg_preds_train, prop_preds_train = self.fit_nuisance_cv(Xtrain, Dtrain, ytrain)
+            self.dr_train = self.calculate_dr_outcomes(Dtrain, ytrain, reg_preds_train, prop_preds_train)
+
+            # Get DR outcomes in validation sample
+            reg_preds_val, prop_preds_val = self.fit_nuisance_train(Xtrain, Dtrain, ytrain, Xval)
+            self.dr_val = self.calculate_dr_outcomes(Dval, yval, reg_preds_val, prop_preds_val)
+
+            self.Dtrain = Dtrain
+
+        else:
+            # Get DR outcomes in validation sample
+            reg_preds_val, prop_preds_val = self.fit_nuisance_cv(Xval, Dval, yval)
+            self.dr_val = self.calculate_dr_outcomes(Dval, yval, reg_preds_val, prop_preds_val)
+
+        # Calculate ATE in the validation sample
+        self.ate_val = self.dr_val.mean(axis=0)
+
+        return self
+
+    def fit_cate(
```

Comment:
I am confused by this method. The evaluation class should not be fitting a cate estimator. It should only be evaluating an already fitted CATE estimator.